lib/scalapackfx.fpp

```diff
@@ -1975,7 +1981,86 @@ contains
     call infog2l(grow, gcol, desc, mygrid%nrow, mygrid%ncol, mygrid%myrow,&
         & mygrid%mycol, lrow, lcol, rsrc, csrc)
 
-  end subroutine scalafx_infog2l
+  end subroutine scalafx_infog2l_single
+
+  !> Maps global positions in a distributed matrix to local one.
+  !!
+  !! \param mygrid  BLACS descriptor.
+  !! \param desc  Descriptor of the distributed matrix.
+  !! \param grow  Global row indices.
+  !! \param gcol  Global column indices.
+  !! \param lrow  Local row indices on output.
+  !! \param lcol  Local column indices on output.
+  !! \param rsrc  Rows of the process owning the local matrix.
+  !! \param csrc  Columns of the process owning the local matrix.
+  !! \param calcAllIndices  Whether to calculate all lrow and lcol,
+  !!     even if the current process does not own them. (default: true)
+  !!
+  subroutine scalafx_infog2l_array(mygrid, desc, grow, gcol,&
+      & lrow, lcol, rsrc, csrc, calcAllIndices)
+    type(blacsgrid), intent(in) :: mygrid
+    integer, intent(in) :: desc(DLEN_)
+    integer, intent(in) :: grow(:), gcol(:)
+    integer, intent(out), dimension(size(grow)) :: lrow, rsrc
+    integer, intent(out), dimension(size(gcol)) :: lcol, csrc
```

Comment:
Please use assumed shape arrays instead. (If you specify the dimension explicitly, querying the size (e.g. in an ASSERT) will be meaningless, as it gives always the same number, even if the actual argument had a different size)
```suggestion
    integer, intent(out) :: lrow(:), rsrc(:)
    integer, intent(out) :: lcol(:), csrc(:)
```